classes/components/forms/FieldRichText.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @file classes/components/form/FieldRichText.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class FieldRichText
+ * @ingroup classes_controllers_form
+ *
+ * @brief A rich signle line text editor field in a form.
+ */
+
+namespace PKP\components\forms;
+
+class FieldRichText extends Field
+{
+    /** @copydoc Field::$component */
+    public $component = 'field-rich-text';
+
+    /** @var array Optional. An assoc array of init properties to pass to TinyMCE */
+    public $init;
+
+    /** @var string Optional. A preset size option. */
+    public $size = 'oneline';
+
+    /** @var string Optional. A preset toolbar configuration. */
+    public $toolbar = 'bold italic underline superscript subscript';
+
+    /** @var int Optional. When a word limit is specified a word counter will be shown */
+    public $wordLimit = 0;
+
+    /** @var array Optional. A list of required plugins. */
+    public $plugins = 'paste';
+
+    /**
+     * @copydoc Field::getConfig()
+     */
+    public function getConfig()
+    {
+        $config = parent::getConfig();
+        $config['toolbar'] = $this->toolbar;
+        $config['plugins'] = $this->plugins;
+
+        if (!empty($this->init)) {
+            $config['init'] = $this->init;
+        }
+
+        if (!empty($this->size)) {
```

Comment:
Currently the size must be `oneline`, so I'd rather we didn't start introducing code that considers it behaving another way. If that becomes necessary later it's easier to add it from scratch rather than trying to figure out whether the partial implementation is safe or used somewhere. Same for `$wordlimit`. If you'd like to e.g. start using this class for other sizes of fields, let's go ahead and implement it now; otherwise wait until we're ready to do that refactor. Partial/preparatory refactors are confusing to figure out after the fact.

Reply:
Make sense , removed .